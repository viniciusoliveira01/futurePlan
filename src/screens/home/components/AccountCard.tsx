import React from 'react';
import { View } from 'react-native';
import { AccountCardProps } from '@src/types/accountCard';
import {
  AccountCardContainer,
  StyledLinearGradient,
  CardContainer,
  CardText,
  StyledIcon,
} from './AccountCard.style';

const AccountCard = ({
  theme,
  title,
  text,
  icon,
  onSimulate,
}: AccountCardProps) => {
  return (
    <AccountCardContainer activeOpacity={0.8} onPress={() => onSimulate()}>
      <StyledLinearGradient
        start={{ x: 0, y: 0 }}
        end={{ x: 1, y: 1 }}
        colors={theme}
      >
        <CardContainer>
          <View>
            <CardText type="name">{title}</CardText>
            <CardText>{text}</CardText>
          </View>
        </CardContainer>

        <StyledIcon name={icon} />
      </StyledLinearGradient>
    </AccountCardContainer>
  );
};

export default AccountCard;
