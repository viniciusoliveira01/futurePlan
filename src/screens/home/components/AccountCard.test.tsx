import React from 'react';
import { render } from '@testing-library/react-native';
import AccountCard from './AccountCard';

describe('Account card', () => {
  it('Should render Account Card', () => {
    const { toJSON } = render(
      <AccountCard
        title="Simule seus investimentos"
        text="e veja quanto você terá investido no futuro"
        icon="ios-arrow-forward"
        theme={['7025FF', '#4e19b2']}
        onSimulate={() => {}}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('Should render card title and text ', () => {
    const { getByText } = render(
      <AccountCard
        title="Card title"
        text="card text"
        icon="ios-arrow-forward"
        theme={['7025FF', '#4e19b2']}
        onSimulate={() => {}}
      />,
    );

    expect(getByText(/Card title/i)).toBeTruthy();
    expect(getByText(/card text/i)).toBeTruthy();
  });
});
