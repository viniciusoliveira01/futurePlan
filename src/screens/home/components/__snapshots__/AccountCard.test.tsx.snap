// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account card Should render Account Card 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          undefined,
          4283308466,
        ],
        "endPoint": Array [
          1,
          1,
        ],
        "locations": undefined,
        "startPoint": Array [
          0,
          0,
        ],
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "color": "#FFF",
          "flexDirection": "row",
          "fontSize": 20,
          "justifyContent": "space-between",
          "marginHorizontal": 20,
          "marginRight": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontSize": 16,
                "fontWeight": "500",
              },
            ]
          }
          type="name"
        >
          Simule seus investimentos
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#FFF",
                "fontSize": 12,
                "fontWeight": "200",
              },
            ]
          }
        >
          e veja quanto você terá investido no futuro
        </Text>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#FFF",
              "fontSize": 20,
              "marginRight": 10,
            },
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
